localstack/services/apigateway/provider.py

```diff
@@ -293,6 +301,74 @@ def put_rest_api(self, context: RequestContext, request: PutRestApiRequest) -> R
         response.setdefault("tags", {})
         return response
 
+    @handler("CreateDomainName")
+    def create_domain_name(
+        self,
+        context: RequestContext,
+        domain_name: String,
+        certificate_name: String = None,
+        certificate_body: String = None,
+        certificate_private_key: String = None,
+        certificate_chain: String = None,
+        certificate_arn: String = None,
+        regional_certificate_name: String = None,
+        regional_certificate_arn: String = None,
+        endpoint_configuration: EndpointConfiguration = None,
+        tags: MapOfStringToString = None,
+        security_policy: SecurityPolicy = None,
+        mutual_tls_authentication: MutualTlsAuthenticationInput = None,
+        ownership_verification_certificate_arn: String = None,
+    ) -> DomainName:
+        if not domain_name:
+            raise BadRequestException("No Domain Name specified")
+
+        store: ApiGatewayStore = get_apigateway_store(context.account_id, context.region)
+        if store.domain_names.get(domain_name):
+            raise ConflictException(f"Domain name with ID {domain_name} already exists")
+
+        # find matching hosted zone
+        zone_id = None
+        route53 = connect_to().route53
+        hosted_zones = route53.list_hosted_zones().get("HostedZones", [])
+        hosted_zones = [hz for hz in hosted_zones if domain_name.endswith(hz["Name"].strip("."))]
+        zone_id = hosted_zones[0]["Id"].replace("/hostedzone/", "") if hosted_zones else zone_id
```

Comment:
question: I'm not too familiar with route53 and the integration with the domain names. Will it always match to a `HostedZone`? What if it doesn't?

Reply:
Not sure I understand the question, maybe something to explore offline. A zone can group many domains. But looking at this code, which was already present in other places I think, it could be done using `list-hosted-zones-by-name`. 

Let me look into it